packages/cardano-services/src/ChainHistory/openApi.json

```diff
@@ -0,0 +1,406 @@
+          "collaterals": {
+            "anyOf": [
+              { "type": "array", "items": { "$ref": "#/components/schemas/TxInput" } },
+              { "$ref": "#/components/schemas/Undefined" }
+            ]
+          },
+          "fee": { "$ref": "#/components/schemas/BigInt" },
+          "inputs": { "type": "array", "items": { "$ref": "#/components/schemas/TxInput" } },
+          "mint": {
+            "anyOf": [{ "$ref": "#/components/schemas/TokenMap" }, { "$ref": "#/components/schemas/Undefined" }]
+          },
+          "outputs": { "type": "array", "items": { "$ref": "#/components/schemas/TxOutput" } },
+          "validityInterval": { "$ref": "#/components/schemas/ValidityInterval" },
+          "withdrawals": {
+            "anyOf": [
+              { "type": "array", "items": { "$ref": "#/components/schemas/Withdrawal" } },
+              { "$ref": "#/components/schemas/Undefined" }
+            ]
+          }
+        }
+      },
+      "Transaction": {
+        "type": "object",
+        "properties": {
+          "auxiliaryData": {
+            "anyOf": [{ "$ref": "#/components/schemas/AuxiliaryData" }, { "$ref": "#/components/schemas/Undefined" }]
+          },
+          "blockHeader": { "$ref": "#/components/schemas/BlockHeader" },
+          "body": { "$ref": "#/components/schemas/TransactionBody" },
+          "id": { "$ref": "#/components/schemas/TxHash" },
+          "implicitCoin": { "$ref": "#/components/schemas/ImplicitCoin" },
+          "index": { "type": "integer" },
+          "txSize": { "type": "integer" },
+          "witness": { "$ref": "#/components/schemas/Witness" }
+        }
+      },
+      "ChainHistoryBlocksBody": {
+        "required": ["args"],
+        "type": "object",
+        "properties": {
+          "args": {
+            "type": "array",
+            "maxLength": 1,
+            "minLength": 1,
+            "items": {
+              "type": "array",
+              "items": { "$ref": "#/components/schemas/BlockHash" }
+            }
+          }
+        }
+      },
+      "ChainHistoryBlocksResponse": {
+        "type": "array",
+        "items": {
+          "$ref": "#/components/schemas/Block"
+        }
+      },
+      "ChainHistoryTxByHashesBody": {
+        "required": ["args"],
+        "type": "object",
+        "properties": {
+          "args": {
+            "type": "array",
+            "maxLength": 1,
+            "minLength": 1,
+            "items": {
+              "type": "array",
+              "items": { "$ref": "#/components/schemas/TxHash" }
+            }
+          }
+        }
+      },
+      "ChainHistoryTxByAddressesBody": {
+        "required": ["args"],
+        "type": "object",
+        "properties": {
+          "args": {
+            "type": "array",
+            "items": {
+              "oneOf": [{ "type": "array", "items": { "$ref": "#/components/schemas/Address" } }, { "type": "number" }]
+            }
+          }
+        }
+      },
```

Comment:
Not sure about this

Reply:
Specifically?:
```
              "oneOf": [{ "type": "array", "items": { "$ref": "#/components/schemas/Address" } }, { "type": "number" }]
```

Yeah, why is there a `number` expected?

Reply:
The number is because `transactionsByAddresses` has a `sinceBlock` optional argument. https://github.com/input-output-hk/cardano-js-sdk/pull/243/files#diff-cc6fbcd6f5f3a6dee2c12a90216ba559ea451cb187699b838f8bed6b600d65c3R339

If we base this on that method arguments like the others the request body should be something like `args: [[addresses], blockNo]`. But I don't think we can validate that with just open api, what I did here would allow `[array, number]`, `[number,array]`, `[array,array]`, `[number, number]`, etc. We could still validate them manually in the provider but the endpoint documentation would be misleading.

Should I change it to an object instead: `args: [{ addresses, blockNo? }]`?
It will be different than the others though as the request body and the provider method arguments don't match. And if you omit `blockNo` the request would look like `args: [{ addresses: Address[] }]`, unlike the others where it's just `args: [[hashes]]`, which is a bit inconsistent I guess

Reply:
> We could still validate them manually in the provider handler but the endpoint documentation would be misleading

Which is an expression of the API :-)  Let's aim for self-describing interfaces at all times 👍 
 
> Should I change it to an object instead: args: [{ addresses, blockNo? }]?

With a slight consistency correction: 
`args: [{ addresses, sinceBlock? }]`

> It will be different than the others though as the request body and the provider method arguments don't match.
...
 which is a bit inconsistent I guess
 
 Please log it as a SPIKE for @mkazlauskas to review, assigned to the production-readiness epic, and we'll press on here.

Reply:
I changed the open api specification to an object in this commit https://github.com/input-output-hk/cardano-js-sdk/pull/243/commits/50827879ec95916cb72e72952de9214a3a35de68, but had to delete these tests here https://github.com/input-output-hk/cardano-js-sdk/pull/243/commits/50827879ec95916cb72e72952de9214a3a35de68#diff-cc6fbcd6f5f3a6dee2c12a90216ba559ea451cb187699b838f8bed6b600d65c3L276 because the open api validation fails when the http provider method makes a request to the http service as the request and method arguments are different now (request: `({ addresses, sinceBlock })`, method: `(addresses, sinceBlock)`). 

Besides that, I don't really see the point of those kinds of tests, the provider already has `chainHistoryHttpProvider.test.ts` and the HttpService is already covered by the tests that use `doChainHistoryRequest`

So I'm not sure what to do, should we revert this commit and leave the request body as an array? Should we change the provider method arguments? Should we leave it like this?

Reply:
>  Should we change the provider method arguments to match the http service?

Yes. I'll make a ticket to review all our Provider interfaces for consistency